utils/docker_entrypoint.sh

```diff
@@ -14,39 +14,31 @@ set -e
 #
 
 # Reasonable defaults if no USER_ID/GROUP_ID environment variables are set.
-if [ -z ${USER_ID+x} ]; then USER_ID=1000; fi
-if [ -z ${GROUP_ID+x} ]; then GROUP_ID=1000; fi
+USER_ID=${USER_ID:-1000}
+GROUP_ID=${GROUP_ID:-1000}
 
 # ccache
 export CCACHE_DIR=/tmp/ccache
 export USE_CCACHE=1
 
 msg="docker_entrypoint: Creating user UID/GID [$USER_ID/$GROUP_ID]" && echo $msg
-groupadd -g $GROUP_ID -r aosp && \
-useradd -u $USER_ID --create-home -r -g aosp aosp
+groupadd -g $GROUP_ID -r aosp ; useradd -u $USER_ID -r -g aosp aosp
 echo "$msg - done"
+echo ""
 
-msg="docker_entrypoint: Copying .gitconfig and .ssh/config to new user home" && echo $msg
-cp /root/.gitconfig /home/aosp/.gitconfig && \
-chown aosp:aosp /home/aosp/.gitconfig && \
-mkdir -p /home/aosp/.ssh && \
-cp /root/.ssh/config /home/aosp/.ssh/config && \
-chown aosp:aosp -R /home/aosp/.ssh &&
+msg="docker_entrypoint: Changing ownership of /tmp/ccache and /aosp..." && echo $msg
+chown -R aosp:aosp /tmp/ccache /aosp
 echo "$msg - done"
+echo ""
 
-msg="docker_entrypoint: Creating /tmp/ccache and /aosp directory" && echo $msg
-mkdir -p /tmp/ccache /aosp
-chown aosp:aosp /tmp/ccache /aosp
+msg="docker_entrypoint: Changing ownership of gitconfig and .ssh/config..." && echo $msg
+chown -R aosp:aosp /home/aosp/.gitconfig /home/aosp/.ssh/
 echo "$msg - done"
-
 echo ""
 
 # Default to 'bash' if no arguments are provided
 args="$@"
-if [ -z "$args" ]; then
-  args="bash"
-fi
 
 # Execute command as `aosp` user
 export HOME=/home/aosp
-exec sudo -u aosp $args
+exec gosu aosp ${args:-"bash"}
```

Comment:
Want to see the reason: 

With sudo -E: 

```
aosp@c1b4674f2f9c:/aosp$ ps waxu 
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  1.0  0.0  47132  3152 pts/0    Ss   00:18   0:00 sudo -u aosp -E bash
aosp        18  0.0  0.0  18172  3200 pts/0    S    00:18   0:00 bash
aosp        21  0.0  0.0  15580  2040 pts/0    R+   00:18   0:00 ps waxu
aosp@c1b4674f2f9c:/aosp$ 
```

That was run on my local host -- not in docker, now with gosu, with gosu -- it becomes PID 1. Not a forked process. 

```
aosp@7e6014c412b9:/aosp$ ps waxu 
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
aosp         1  1.7  0.0  18172  3316 pts/0    Ss   00:16   0:00 bash
aosp        24  0.0  0.0  15580  2040 pts/0    R+   00:16   0:00 ps waxu
aosp@7e6014c412b9:/aosp$ 
```